pkg/kubelet/container/runtime.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package container
+
+import (
+	"github.com/fsouza/go-dockerclient"
+)
+
+// ContainerRuntimeInterface is the abstract interface for container runtime.
+type ContainerRuntimeInterface interface {
+	ListContainers(options docker.ListContainersOptions) ([]docker.APIContainers, error)
```

Comment:
Almost every call contains a parameter from go-dockerclient: ListContainerOptions, Container, CreateContainerOptions, ... Really those parameter structs have to be part of the interface. 

I'm fine with this PR as a first step, but you should ask fsouza whether he'd accept a major refactoring of the library to target multiple container runtimes at that level. Otherwise, we should consider forking that library. 

For now, a simple change would be to use an alias for that package here that is more generic.